File: ChatGPT/Python/FactoryMethod/round3/source/pastry_factory.py
from abc import ABC, abstractmethod

# Abstract Product
class Pastry(ABC):
    @abstractmethod
    def get_description(self):
        pass

# Concrete Products
class Croissant(Pastry):
    def get_description(self):
        return "Croissant"

class Bun(Pastry):
    def get_description(self):
        return "Bun"

class CustardBread(Pastry):
    def get_description(self):
        return "Custard Bread"

class ChocolateCake(Pastry):
    def get_description(self):
        return "Chocolate Cake"

class StrawberryCake(Pastry):
    def get_description(self):
        return "Strawberry Cake"

class CoconutCake(Pastry):
    def get_description(self):
        return "Coconut Cake"

class ChocolateChipCookies(Pastry):
    def get_description(self):
        return "Chocolate Chip Cookies"

class OatCookies(Pastry):
    def get_description(self):
        return "Oat Cookies"

# Abstract Factory
class PastryFactory(ABC):
    @abstractmethod
    def create_pastry(self):
        pass

# Concrete Factories
class CroissantFactory(PastryFactory):
    def create_pastry(self):
        return Croissant()

class BunFactory(PastryFactory):
    def create_pastry(self):
        return Bun()

class CustardBreadFactory(PastryFactory):
    def create_pastry(self):
        return CustardBread()

class ChocolateCakeFactory(PastryFactory):
    def create_pastry(self):
        return ChocolateCake()

class StrawberryCakeFactory(PastryFactory):
    def create_pastry(self):
        return StrawberryCake()

class CoconutCakeFactory(PastryFactory):
    def create_pastry(self):
        return CoconutCake()

class ChocolateChipCookiesFactory(PastryFactory):
    def create_pastry(self):
        return ChocolateChipCookies()

class OatCookiesFactory(PastryFactory):
    def create_pastry(self):
        return OatCookies()

File: ChatGPT/Java/FactoryMethod/round3/source/PastryFactory.java
package ChatGPT.Java.FactoryMethod.round3.source;

public class PastryFactory extends ProductFactory {
     @Override
     public Product createProduct(String type) {
         switch (type) {
             case "Croissant":
                 return new Croissant();
             case "Bun":
                 return new Bun();
             case "Custard Bread":
                 return new CustardBread();
             case "Chocolate Cake":
                 return new ChocolateCake();
             case "Strawberry Cake":
                 return new StrawberryCake();
             case "Coconut Cake":
                 return new CoconutCake();
             case "Chocolate Chip Cookies":
                 return new ChocolateChipCookies();
             case "Oat Cookies":
                 return new OatCookies();
             default:
                 throw new IllegalArgumentException("Unknown pastry type");
         }
     }
 }
 
